src/command.rs

```diff
@@ -54,9 +54,17 @@ impl Command {
       Self::Operation(operation) => state.operation = *operation,
       Self::Print => state.print()?,
       Self::Repl => {
-        for result in BufReader::new(io::stdin()).lines() {
-          let line = result?;
-          match line.trim().parse::<Self>() {
+        let history = tilde("~/.degenerate_history").to_string();
```

Comment:
I'd personally prefer the `directories` crates, but just for aesthetic reasons:

https://docs.rs/directories/latest/directories/struct.UserDirs.html#method.home_dir

TIOLI